### ISSUE TITLE
Failed to create CoreCLR, HRESULT: 0x80004005 with WSL1

### ISSUE THREAD

**Issue by josephmoresena** [dotnet/runtime#72192]
### Description

I'm trying to build CoreCLR on Ubuntu using WSL1. The system displays the following output:
`
__DistroRid: linux-x64
Failed to create CoreCLR, HRESULT: 0x80004005
Build failed with exit code 137. Check errors above.
`

### Reproduction Steps

In runtime repo directory execute:
`./build.sh -subset clr`

### Expected behavior

Start CoreCLR compiling.

### Actual behavior

Build failed with exit code 137. Check errors above.

**Comment by EgorBo** [comment]
You need to inspect the full log, e.g. make sure you installed all the needed dependencies from https://github.com/dotnet/runtime/blob/main/docs/workflow/requirements/linux-requirements.md
Also, check your cmake version

**Comment by josephmoresena** [comment]
> You need to inspect the full log, e.g. make sure you installed all the needed dependencies from https://github.com/dotnet/runtime/blob/main/docs/workflow/requirements/linux-requirements.md Also, check your cmake version

This is my current configuration:
`
Reading package lists... Done
Building dependency tree
Reading state information... Done
Note, selecting 'python-is-python2' instead of 'python'
Note, selecting 'libunwind-dev' instead of 'libunwind8-dev'
cmake is already the newest version (3.16.3-1ubuntu1).
gettext is already the newest version (0.19.8.1-10build1).
liblttng-ust-dev is already the newest version (2.11.0-1).
libnuma-dev is already the newest version (2.0.12-1).
libunwind-dev is already the newest version (1.2.1-9build1).
libunwind8 is already the newest version (1.2.1-9build1).
clang-9 is already the newest version (1:9.0.1-12).
liblldb-6.0-dev is already the newest version (1:6.0.1-14).
lldb-6.0 is already the newest version (1:6.0.1-14).
llvm-9 is already the newest version (1:9.0.1-12).
ninja-build is already the newest version (1.10.0-1build1).
python-is-python2 is already the newest version (2.7.17-4).
build-essential is already the newest version (12.8ubuntu1.1).
curl is already the newest version (7.68.0-1ubuntu2.12).
git is already the newest version (1:2.25.1-1ubuntu3.5).
libicu-dev is already the newest version (66.1-2ubuntu2.1).
libkrb5-dev is already the newest version (1.17-6ubuntu4.1).
libssl-dev is already the newest version (1.1.1f-1ubuntu2.16).
zlib1g-dev is already the newest version (1:1.2.11.dfsg-2ubuntu1.3).
0 upgraded, 0 newly installed, 0 to remove and 40 not upgraded
`

**Comment by jkotas** [comment]
We are not doing any testing on WSL1. We should document that it is not supported. The current dotnet/runtime on WSL1 is likely completely broken.

**Comment by josephmoresena** [comment]
> We are not doing any testing on WSL1. We should document that it is not supported. The current dotnet/runtime on WSL1 is likely completely broken.

I see... So sad read that. All my NativeAOT Linux tests are done in WSL1. WSL2 is not really a solution for me as sometimes the resource consumption can be quite extensive.

**Comment by MichalStrehovsky** [comment]
There is a workaround in https://github.com/dotnet/runtime/issues/70799#issuecomment-1158427454

**Comment by MichalPetryka** [comment]
> There is a workaround in [#70799 (comment)](https://github.com/dotnet/runtime/issues/70799#issuecomment-1158427454)

Wasn't #70912 supposed to fix W^X under WSL1?

**Comment by MichalStrehovsky** [comment]
> > There is a workaround in [#70799 (comment)](https://github.com/dotnet/runtime/issues/70799#issuecomment-1158427454)
> 
> Wasn't #70912 supposed to fix W^X under WSL1?

The repo builds with .NET 7 Preview 5 that doesn't have the fix.

**Comment by josephmoresena** [comment]
Thanks a lot. I'm now compilling CoreCLR on WSL1 with the workaround. 
Will WSL1 become bad enough (in the medium term) for no use .NET with it?

**Comment by MichalStrehovsky** [comment]
> Will WSL1 become bad enough (in the medium term) for no use .NET with it?

As Jan said, we're not testing with it.

That said, I use it in my local workflows for similar reasons as you. I only have 16 GB of RAM and a small SSD. WSL2 is very much not a good fit for that.

**Comment by josephmoresena** [comment]
> > Will WSL1 become bad enough (in the medium term) for no use .NET with it?
> 
> As Jan said, we're not testing with it.
> 
> That said, I use it in my local workflows for similar reasons as you. I only have 16 GB of RAM and a small SSD. WSL2 is very much not a good fit for that.

Yes, thats why I ask that. So.. We just have to take advantage of it as much as possible. 
It's really too bad that WSL1 has been frozen in time. I just loved it when I could see all the possibilities. It really makes me run away from dual booting and heavy VM (including WSL2), even bare-metal Linux (I only use it now in my personal server).

**Comment by mangod9** [comment]
Assume this is ok to close, now that there is a workaround?
